sources/rest_api/typing.py

```diff
@@ -0,0 +1,109 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    TypedDict,
+    Union,
+    Literal,
+)
+
+from dlt.common import jsonpath
+from dlt.common.typing import TSortOrder
+from dlt.extract.items import TTableHintTemplate
+from dlt.extract.incremental.typing import LastValueFunc
+
+from .paginators import BasePaginator
+from .auth import AuthConfigBase
+
+from dlt.common.schema.typing import (
+    TColumnNames,
+    # TSchemaContract,
+    TTableFormat,
+    TTableSchemaColumns,
+    TWriteDisposition,
+)
+
+PaginatorConfigDict = Dict[str, Any]
+PaginatorType = Union[Any, BasePaginator, str, PaginatorConfigDict]
+
+HTTPMethodBasic = Literal["get", "post"]
+HTTPMethodExtended = Literal["put", "patch", "delete"]
```

Comment:
I think just for the sake of completeness and maybe to support weird APIs should we also add `options` and `head` methods?